File: Dante1/Module1.07/Zadanie7.13/main.c
#include <stdio.h>
#include <math.h>

int goldbach(int num);
int czy_pierwsza(int num);

int main()
{
    int num1, num2;

    printf("Enter num1: ");
    int check1 = scanf("%i",&num1);
    if(check1 != 1)
    {
        printf("Incorrect input");
        return 1;
    }

    printf("Enter num2: ");
    int check2 = scanf("%i",&num2);
    if(check2 != 1)
    {
        printf("Incorrect input");
        return 1;
    }

    if(num2 < num1)
    {
        printf("Incorrect data input");
        return 2;
    }

    printf("\n");
    for(int i = num1; i <= num2; i++)
    {
        if(i % 2 == 0)
        {
            printf("%i: ",i);
            goldbach(i);
            printf("\n");
        }
    }

    return 0;
}

int goldbach(int num)
{
    int pointer = 0;
    if (num % 2 != 0 || num < 4)
    {
        return 0;
    }
    for (int i = 2; i < num; i++)
    {
        if (czy_pierwsza(i) == 1)
        {
            int j = num - i;
            if (czy_pierwsza(j) == 1)
            {
                printf("(%i, %i) ", i, j);
                pointer = 1;
            }
        }
    }

    return pointer;
}

int czy_pierwsza(int num)
{
    if (num < 2)
    {
        return 0;
    }

    int sqrt_num = sqrt(num);
    for (int i = 2; i <= sqrt_num; i++)
    {
        if (num % i == 0)
        {
            return 0;
        }
    }

    return 1;
}

File: Dante1/Module1.07/Zadanie7.13/main.rs
fn prime(num: i32) -> i32 {
    if num < 2 {
        return 0;
    }

    let sqrt_num = i32::isqrt(num);
    for i in 2..=sqrt_num {
        if num % i == 0 {
            return 0;
        }
    }

    1
}

fn goldbach(num: i32) -> i32 {
    let mut point: i32 = 0;

    if num % 2 != 0 || num < 4 {
        return 0;
    }

    for i in 2..num {
        if prime(i) == 1 {
            let mut j: i32 = num - i;
            if prime(j) == 1 {
                println!("({}, {})", i, j);
                point = 1;
            }
        }
    }

    point
}

fn main() {
    let mut input: String = String::new();

    println!("Enter: ");
    std::io::stdin().read_line(&mut input).expect("err");
    let mut num1: i32 = input.trim().parse().expect("err");

    input.clear();

    println!("Enter: ");
    std::io::stdin().read_line(&mut input).expect("err");
    let mut num2: i32 = input.trim().parse().expect("err");

    if num2 < num1 {
        panic!("Incorrect input");
    }

    for i in num1..=num2 {
        if i % 2 == 0 {
            print!("I: {}", i);
            goldbach(i);
            println!();
        }
    }
}
